src/contracts/atlas/AtlasVerification.sol

```diff
@@ -589,34 +560,64 @@ contract AtlasVerification is EIP712, DAppIntegration, AtlasConstants {
         return ValidCallsResult.Valid;
     }
 
-    /// @notice Generates the hash of a UserOperation struct.
-    /// @param userOp The UserOperation struct to hash.
-    /// @return userOpHash The hash of the UserOperation struct.
-    function _getUserOpHash(UserOperation calldata userOp) internal pure returns (bytes32 userOpHash) {
-        userOpHash = keccak256(
-            abi.encode(
-                USER_TYPEHASH,
-                userOp.from,
-                userOp.to,
-                userOp.value,
-                userOp.gas,
-                userOp.maxFeePerGas,
-                userOp.nonce,
-                userOp.deadline,
-                userOp.dapp,
-                userOp.control,
-                userOp.callConfig,
-                userOp.sessionKey,
-                userOp.data
-            )
-        );
+    /// @notice Generates the payload hash of a UserOperation struct used in signatures.
+    /// @param userOp The UserOperation struct to generate the payload for.
+    /// @return payload The hash of the UserOperation struct for use in signatures.
+    function getUserOperationPayload(UserOperation calldata userOp) public view returns (bytes32 payload) {
+        payload = _getUserOperationHash(userOp, false);
     }
 
-    /// @notice Generates the hash of a UserOperation struct.
-    /// @param userOp The UserOperation struct to hash.
-    /// @return payload The hash of the UserOperation struct.
-    function getUserOperationPayload(UserOperation calldata userOp) public view returns (bytes32 payload) {
-        payload = _hashTypedDataV4(_getUserOpHash(userOp));
+    /// @notice Generates the hash of a UserOperation struct used for inter-operation references.
+    /// @param userOp The UserOperation struct to generate the hash for.
+    /// @return hash The hash of the UserOperation struct for in inter-operation references.
+    function getUserOperationHash(UserOperation calldata userOp) public view returns (bytes32 hash) {
+        uint32 callConfig = IDAppControl(userOp.control).CALL_CONFIG();
```

Comment:
CallConfig is already a property in UserOperation struct, so we don't need to make another call to retrieve it. I think it was only added recently though, so this improvement may apply to other areas of the code e.g. `_getOrCreateExecutionEnvironment()` in Factory.sol

Reply:
ee96b62fe2b0379531bcabbb5d14f8d61c3a1e37